Docs: Add code examples section (#22974)
WordPress/gutenberg: docs/contributors/document.md

This way they will be properly handled in all three aforementioned contexts.

### Code Examples

> The code example in markdown should be wrapped in three tick marks <code>```</code> and can additionally include a language specifier. See this [Github documentation around fenced code blocks](https://help.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks).

Reviewer: ```suggestion
The code example in markdown should be wrapped in three tick marks <code>```</code> and can additionally include a language specifier. See this [GitHub documentation around fenced code blocks](https://help.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks).
```

A unique feature to the Gutenberg documentation is the `codetabs` toggle, this allows two versions of code to be shown at once. This is used for showing both `ESNext` and `ES5` code samples. For example, [on this block tutorial page](/docs/designers-developers/developers/tutorials/block-tutorial/block-controls-toolbar-and-sidebar.md).

Here is an example `codetabs` section:

		{% codetabs %}
		{% ESNext %}
		```js
			// ESNext code here
		```
		{% ES5 %}
		```js
			// ES5 code here
		```

> 		{% end %}

Reviewer: Was there a reason for using two tabs? Usually one tab (four spaces) is enough to create a code block.

Author: Overly, aggressive tabbing. :-)

> The preferred format for code examples is ESNext, which should also be the default viewed, when placed first in source it will show as the default.

Reviewer: I think this reads better when split into two sentences. Something like this:

```suggestion
The preferred format for code examples is ESNext, which should also be the default viewed. The example placed first in source will be shown as the default.
```
